BeVolt/BSP/Simulator/DataGeneration/CAN.py

```diff
@@ -56,12 +56,13 @@ def Send_Message(id, message, length):
         fcntl.flock(csvfile.fileno(), fcntl.LOCK_UN)
 
 def Get_CAN_Info():

```

Comment:
Wait for long enough, and this function will raise a StopIteration exception. This crashes the simulator. I've hit it a few times, usually after under one minute of running the simulator with the BSP_CAN test. 

This exception is standard of iterators: they raise this when they are out of data. My guess would be that instead of checking whether the file is non-empty, you need to check whether there is any data that has not already been read.

Reply:
I believe that I resolved this problem (see ad09785610d70cc50cd1aa84a7f93169a908e368).